File: app/src/main/java/re/snapwi/orientation/io/Value.java
package re.snapwi.orientation.io;

import android.os.Parcel;
import android.os.Parcelable;

public class Value implements Parcelable {
  String joke;
  String id;

  public String getJoke() {
    return joke;
  }

  public String getId() {
    return id;
  }

  Value(Parcel in) {
    joke = in.readString();
    id = in.readString();
  }

  @Override public int describeContents() {
    return 0;
  }

  @Override public void writeToParcel(Parcel dest, int flags) {
    dest.writeString(joke);
    dest.writeString(id);
  }

  @SuppressWarnings("unused") public static final Creator<Value> CREATOR = new Creator<Value>() {
    @Override public Value createFromParcel(Parcel in) {
      return new Value(in);
    }

    @Override public Value[] newArray(int size) {
      return new Value[size];
    }
  };
}

File: app/src/main/kotlin/re/snapwi/orientation/io/Value.kt
package re.snapwi.orientation.io

import android.os.Parcel
import android.os.Parcelable

public class Value : Parcelable {
  public var joke: String? = null
  public var id: String? = null

  public constructor(parcel: Parcel) {
    joke = parcel.readString()
    id = parcel.readString()
  }

  override fun describeContents(): Int = 0

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeString(joke)
    dest.writeString(id)
  }

  companion object {
    public val CREATOR: Parcelable.Creator<Value> = object : Parcelable.Creator<Value> {
      override fun createFromParcel(parcel: Parcel): Value = Value(parcel)
      override fun newArray(size: Int): Array<Value?> = arrayOfNulls(size)
    }
  }
}
